plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md

```diff
@@ -0,0 +1,388 @@
+					],
+					[
+						'value' => 'friend',
+						'label' => 'From a friend'
+					],
+					[
+						'value' => 'other',
+						'label' => 'Other'
+					],
+				]
+			)
+		);
+	}
+);
+```
+
+This results in the additional information section being rendered like so:
+
+<p>
+	<img width="701" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/bbe17ad0-7c7d-419a-951d-315f56f8898a"><br />
+	<em>The select input before being focused</em>
+</p>
+
+<p>
+	<img width="724" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/bd943906-621b-404f-aa84-b951323e25d3"><br />
+	<em>The select input when focused</em>
+</p>
+
+If it is undesirable to force the shopper to select a value, providing a value such as "None of the above" may help.
+
+## Validation
+
+It is possible to add custom validation to any registered additional checkout field as long as you know its namespace and ID.
+
+To do so, use the `woocommerce_blocks_validate_additional_field_{namespace}` filter.
+
+For example to apply validation to the example text field above use the `woocommerce_blocks_validate_additional_field_namespace/gov-id` filter.
+
+This filter receives the following arguments, in order.
+
+
+| Argument        | Type       | Description                                                                                                                                                                                                                                                     |
+|-----------------|------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `$error`        | `WP_Error` | A `WP_Error` object. Initially it is empty. This filter should add errors to it using the [`add`](https://developer.wordpress.org/reference/classes/wp_error/add/) method. The returned value of this filter must be the _same_ WP_Error that was passed to it. |
+| `$field_value`  | `mixed`      | The current value of the field, i.e. what the user entered in the checkout form.                                                                                                                                                                                |
+| `$field_schema` | `array`      | The schema of the field. This is what is registered with Store API. This is useful to know what values are valid for select fields.                                                                                                                             |
+| `$key`          | `string`     | The field's key (composed of namespace/id)                                                                                                                                                                                                                      |
```

Comment:
I'm raising this in slack but:

- I don't see much value in field key here.
- I'm torn a bit about field_schema
- We do really need the order, customer, and/or request here.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/43789 and docs updated in fd5ce3e63cb044a1eedbba78e6337cb08f39648f.

> I'm torn a bit about field_schema

Required to allow extensions to see other possible options on the `select` fields, I think it's potentially useful to expose, but I would also be OK not exposing it. I don't have a strong preference either way. It can be re-added later if we feel it is needed? Should we remove it in https://github.com/woocommerce/woocommerce/pull/43789?

Reply:
For now i'm going to remove this, we can re-add it later if needed. Better to start small then add rather than try and reduce later 👌🏼  will open a PR.

Reply:
Removed in https://github.com/woocommerce/woocommerce/pull/44012